source/common/stats/stats_impl.h

```diff
@@ -160,21 +186,28 @@ class RawStatDataAllocator {
  */
 class MetricImpl : public virtual Metric {
 public:
-  MetricImpl(const std::string& name) : name_(name) {}
+  MetricImpl(const std::string& name, std::string&& tag_extracted_name, std::vector<Tag>&& tags)
+      : name_(name), tag_extracted_name_(std::move(tag_extracted_name)), tags_(std::move(tags)) {}
 
   const std::string& name() const override { return name_; }
+  const std::string& tagExtractedName() const override { return tag_extracted_name_; }
+  const std::vector<Tag>& tags() const override { return tags_; }
 
 private:
   const std::string name_;
+  const std::string tag_extracted_name_;
```

Comment:
nit: can we make all of these const?

Reply:
?  These members are already const.  Was this meant to point at something else?

Reply:
Oops sorry nevermind mistake.